NExtends/Primitives/DateTimes/DateTime.extensions.cs

```diff
@@ -154,7 +154,11 @@ public static int GetDifferenceInFullYearsWith(this DateTime firstDate, DateTime
 			DateTime zeroTime = new DateTime(1, 1, 1);
 
 			var timeSpan = firstDate - secondDate;
-			var yearResult = (zeroTime + timeSpan).Year - 1; // because we start at year 1 for the Gregorian calendar, we must subtract a year here.
+            if (timeSpan < System.TimeSpan.Zero)
+            {
+                throw new ArgumentOutOfRangeException();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use a constructor overloads that allows a more meaningful exception message to be provided. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)